MinimalLifts.m

```diff
@@ -0,0 +1,231 @@
+MaximalOrder:=function(G)
+  x:=[z:z in Generators(G)];
+a:=x[1];
+n:=Order(G);
+i:=1;
+while Order(a) ne n do
+	   i:=i+1;
+a:=a*x[i];
+end while;
+return a;
+end function;
+
+
+CentralProductPC:=function(n,H,Z)
+  T:=H meet Z;
+k1:=Order(T);
+if Order(Z) eq 1 then
+return H;
+end if;
+p:=Order(H);
+k2:=n*k1;
+k:=k2/p;
+k:=Numerator(k);
+C:=PCGroup(CyclicGroup(k));
+B, inc:=DirectProduct(H, C);
+x:=MaximalOrder(Z);
+y:=MaximalOrder(C);
+z:=Order(Z);
+a:=z/k1;
+a:=Numerator(a);
+b:=k/k1;
+b:=Numerator(b);
+N:=sub<B|inc[1](x^a)*inc[2](y^b)>;
+return B / N;
+end function;
+
+
+CentralProductPerm:=function(n,H,Z)
+  T:=H meet Z;
+k1:=Order(T);
+if Order(Z) eq 1 then
+return H;
+end if;
+p:=Order(H);
+k2:=n*k1;
+k:=k2/p;
+k:=Numerator(k);
+C:=CyclicGroup(k);
+B, inc:=DirectProduct(H, C);
+x:=MaximalOrder(Z);
+y:=MaximalOrder(C);
+z:=Order(Z);
+a:=z/k1;
+a:=Numerator(a);
+b:=k/k1;
+b:=Numerator(b);
+N:=sub<B|inc[1](x^a)*inc[2](y^b)>;
+return B / N;
+end function;
+
+
+CentralExtensionsIsMinimalLift:=function(G)
+  Z:=Center(G);
+n:=Order(G);
+S:=Subgroups(G);
+k:=#S;
+k:=k-1;
+if IsSolvable(G) then
+for j in [k..1 by -1] do
+      H:=S[j]`subgroup;
+if IsIsomorphicSolubleGroup(CentralProductPC(n, H, Z), G) then
+return false;
+end if;
+end for;
+ else
+   for j in [k..1 by -1] do
+	 H:=S[j]`subgroup;
+if IsIsomorphic(CentralProductPerm(n, H, Z), G) then
+return false;
+end if;
+end for;
+end if;
+return true;
+end function;
+
+
+TrivialActionConstruction:=function(G)
+  n:=Ngens(G);
+x:=[];
+for i in [1..n] do
+	x[i]:=[1];
+end for;
+return x;
+end function;
+
+
+PermGroup:=function(G)
+  H:=sub<G|Id(G)>;
```

Comment:
You can use either `MinimalDegreePermutationRepresentation` or `MyQuotient` from https://github.com/roed314/FiniteGroups/blob/master/Code/Quotients.m.